src/ECER.Clients.Api/InviteLinks/InviteLinksEndpoints.cs

```diff
@@ -0,0 +1,34 @@
+﻿using ECER.Managers.Admin.Contract.InviteLinks;
+using ECER.Utilities.Hosting;
+using MediatR;
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace ECER.Clients.Api.References;
+
+public class InviteLinksEndpoints : IRegisterEndpoints
+{
+  public void Register(IEndpointRouteBuilder endpointRouteBuilder)
+  {
+    endpointRouteBuilder.MapPost("/api/invitelinks", async Task<Results<Ok<GenerateInviteLinkResponse>, BadRequest<string>>> (
+      GenerateInviteLinkRequest request,
+      HttpContext httpContext,
+      IMediator messageBus,
+      IConfiguration configuration,
+      CancellationToken ct) =>
+    {
+      var referenceLinkResponse = await messageBus.Send(new GenerateInviteLinkCommand(request.portalInvitation, request.inviteType, request.validDays), ct) as GenerateInviteLinkCommandResponse;
+      string baseUrl = configuration["PortalApp:BaseUrl"]!;
```

Comment:
You probably want to use the options pattern and register in the configure component of the engine, then inject to the instance